src/analysis/fourier/poisson_summation.lean

```diff
@@ -0,0 +1,114 @@
+/-
+Copyright (c) 2023 David Loeffler. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: David Loeffler
+-/
+import analysis.fourier.add_circle
+import analysis.fourier.fourier_transform
+
+/-!
+# Poisson's summation formula
+
+We prove Poisson's summation formula `∑ (n : ℤ), f n = ∑ (n : ℤ), 𝓕 f n`, where `𝓕 f` is the
+Fourier transform of `f`, under the following hypotheses:
+* `f` is a continuous function `ℝ → ℂ`.
+* The sum `∑ (n : ℤ), 𝓕 f n` is convergent.
+* For all compacts `K ⊂ ℝ`, the sum `∑ (n : ℤ), sup { ‖f(x + n)‖ | x ∈ K }` is convergent.
+
+## TODO
+
+* Show that the conditions on `f` are automatically satisfied for Schwartz functions.
+-/
+
+noncomputable theory
+
+open function (hiding comp_apply) complex real set (hiding restrict_apply)
+  topological_space filter measure_theory
+
+open_locale real big_operators filter fourier_transform
+
+local attribute [instance] real.fact_zero_lt_one
+
+open continuous_map
+
+/-- The key lemma for Poisson summation: the `m`-th Fourier coefficient of the periodic function
+`∑' n : ℤ, f (x + n)` is the value at `m` of the Fourier transform of `f`. -/
+lemma real.fourier_coeff_tsum_comp_add {f : C(ℝ, ℂ)}
+  (hf : ∀ (K : compacts ℝ), summable (λ n : ℤ, ‖(f.comp (continuous_map.add_right n)).restrict K‖))
+  (m : ℤ) :
+  fourier_coeff (periodic.lift $ f.periodic_tsum_comp_add_zsmul 1) m = real.fourier_integral f m :=
```

Comment:
Can you use the notation `𝓕` for `real.fourier_integral`, here and elsewhere?

Reply:
Sure. I took it out because I wondered if it it might be preferable to reserve it for the (yet-to-be-defined) Fourier transform on Lp spaces, etc; but I'm happy to add it back in again.